### ISSUE TITLE
COOP/COEP SharedArrayBuffer headers not correctly implemented for Safari

### ISSUE THREAD

**Issue by amirrajan** [itchio/itch.io#1461]
It seems that Safari isn’t happy with the COOP/COEP headers Itch is serving up. The following game works correctly on a local server I’ve set up, but when deployed to Itch the game fails to load: https://amirrajan.itch.io/a-car-that-turns

**Comment by amirrajan** [comment]
The game correctly loads on Chrome and Firefox, but not Safari/Safari Mobile.

**Comment by icculus** [comment]
It’s notable that the COOP/COEP headers are correctly set for index.html, but Safari/iOS seems to want it set on other files, also…perhaps the js or wasm files, too?

Nothing else, including Safari on macOS, has this problem, so it’s _probably_ Apple’s bug, but a workaround of some sort is needed here.

**Comment by marcheiligers** [comment]
Yep, I've got this problem with my game too, which is a real pity since my game is made to be played on mobile and this problem effectively means people with iPhones are unable to play it.

**Comment by icculus** [comment]
I played around with this on a personal Apache server, and the only files that need both COOP and COEP headers are index.html and dragonruby-wasm.worker.js, to make Safari happy.

Both files currently have the COEP header but they both must have the COOP header too, or Safari won't play ball.

Presumably what is happening is that the main thread is loading index.html, which sees the proper headers, so it's cool with our other .js and .wasm files, but the .worker.js is being loaded in a Web Worker and Safari wants to see these headers set for each worker...? This is just a guess.

We don't control this worker.js file; Emscripten produces it (and _requires it_) for multithreaded apps.

We've got a game jam coming up in a week, and as it stands, no one using our engine can publish web builds to itch.io that work on Macs or iPhones because the web server is setting _one_ required response header and not _two_.

Can we get a response to this problem? I'll absolutely settle for "you're doing it wrong, load your stuff this way instead," and I'll make the appropriate changes in the engine, but I think we just need the server to set the extra header.

**Comment by icculus** [comment]
I hate to be rude and keep pinging, but we are really running out of time now; @leafo, is this something that can trivially be tweaked on the server-side, or is there something _we_ can tweak to make this itch.io-friendly?

We're experimenting with sticking a ServiceWorker in the pipeline on our end, and while this resolves the issue on a standard webpage that doesn't supply the proper headers, we haven't quite worked out the magic to make this work when embedded on itch yet...but having the headers set correctly is the most-efficient and expedient solution here.

**Comment by leafo** [comment]
Did some tests with safari today and I don't see any issues with whats currently deployed https://leafo2.itch.io/hello-world 

Closing out this issue.
